faiss/IndexIVFPQ.cpp

```diff
@@ -433,10 +434,10 @@ void initialize_IVFPQ_precomputed_table(
     }
 
     if (use_precomputed_table == 0) { // then choose the type of table
-        if (quantizer->metric_type == METRIC_INNER_PRODUCT) {
+        if (!(quantizer->metric_type == METRIC_L2 && by_residual)) {
             if (verbose) {
                 printf("IndexIVFPQ::precompute_table: precomputed "
-                       "tables not needed for inner product quantizers\n");
+                       "tables not needed only if L2 quantizer and by_residual is enabled\n");
```

Comment:
please change this to " tables needed only for  L2 metric and by_residual is enabled" and make sure the line is < 80 characters.

Reply:
Done.